### PR TITLE
Introduce user set creation and options for controlling dynamic classification results

### PR DESCRIPTION
**Pull request by jeverley**
@ldir-EDB0 let me know if you've any suggestions/tweaks before I merge into main.

### CODE REVIEW

File: etc/init.d/dscpclassify [hunk 1]
```diff
@@ -3,11 +3,92 @@
...
+		esac
+	done
+}
+
+sort_ip() {
+	for i in $1; do
+		case $i in
+		[0-9]*.[0-9]*.[0-9]*.*[0-9]) ipv4="$ipv4 $i" ;;
+		*:*:*) ipv6="$ipv6 $i" ;;
+		@*[a-zA-Z0-9]) ipset="$ipset $i" ;;
```
Comment (ldir-EDB0) [inline etc/init.d/dscpclassify:88]: I think you should include "-" and "_" as permitted set name characters.  They're nftables legal and I commonly use a construction "setname_4" and "setname_6" for ipv4/v6 specific sets.

Comment (jeverley) [reply etc/init.d/dscpclassify]: Good spot @ldir-EDB0, I've just committed some updates to the validations that should allow this now (as well as making them more robust overall).

File: etc/init.d/dscpclassify [hunk 2]
```diff
@@ -45,173 +127,229 @@ iifzone() {
...
+
+	config_get_bool constant "$1" constant
+	config_get element "$1" element
+	config_get_bool interval "$1" interval
+	config_get family "$1" family
+	config_get name "$1" name
+	config_get timeout "$1" timeout
+	config_get type "$1" type
+
+	if [ -z "$famiy$type" ] || [ -n "$family" ] && [ -n "$type" ]; then
```
Comment (ldir-EDB0) [inline etc/init.d/dscpclassify:307]: typo - should be 'family'

Comment (jeverley) [reply etc/init.d/dscpclassify]: Fixed in [6c4cef4](https://github.com/jeverley/dscpclassify/pull/5/commits/6c4cef41e12a1fee9f89b4eb1d0c5b0df99ca937)

**Pull request merged**